### PR TITLE
feat(content-releases): add functionality to publish release button

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

Link publish button in release details view with its endpoint.

### How to test it?

You should have an EE license and the feature enabled by setting contentReleases as enabled on the feature file.
```
// myproject/config/features.js
module.exports = ({ env }) => ({
  future: {
    contentReleases: true
  },
});
```

1. Create a new release and add entries to it
2. Publish this release clicking on the "Release" button on the top right
3. You should see a success message  and now the release is on "only-view" mode
4. All the entries related should be published/unpublished

### CODE REVIEW

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx [hunk 1]
```diff
@@ -169,86 +200,100 @@ export const ReleaseDetailsLayout = ({
...
-                {formatMessage({
-                  id: 'content-releases.header.actions.release',
-                  defaultMessage: 'Release',
-                })}
-              </Button>
-            </Flex>
+                    {formatMessage({
+                      id: 'content-releases.header.actions.release',
+                      defaultMessage: 'Release',
+                    })}
```
Comment (markkaylor) [inline packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:292]: I think we agreed and the designs have been updated with Publish now

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx [hunk 2]
```diff
@@ -398,11 +449,24 @@ const ReleaseDetailsBody = () => {
...
+                    {release.releasedAt ? (
+                      <Typography>
+                        {formatMessage(
+                          {
+                            id: 'content-releases.page.ReleaseDetails.table.action-published',
+                            defaultMessage:
+                              'This entry was {isPublish, select, true {published} other {unpublished}}.',
+                          },
+                          { isPublish: type === 'publish' }
+                        )}
```
Comment (markkaylor) [inline packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:461]: I don't see the view for "done" releases in the designs. Do you know where it is? For some reason I thought we were just going to show the buttons but disable them 🤔

Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:461]: Found it on the designs! I think you just need the bold around the published unpublished. Rémi did that somewhere I'll find it.

Comment (Feranchz) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:461]: Yes, I was just to leave a comment, I tried a lot of ways to bold that but nothing works 😢 Please let me know if you found an example

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx (hunk 1 above)
Comment (Feranchz) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:292]: You're right, completely forgot about it, thanks!

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx (hunk 2 above)
Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:461]: I have this example where a custom component is provided for the b tag, but here I don't think it should be necessary:

https://github.com/strapi/strapi/blob/523f211d2b0a060359d7d02674b39cba499796d6/packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/ConfirmBulkActionDialog/index.js#L130-L137

I'll try to find an example without custom component

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx [hunk 3]
```diff
@@ -169,86 +200,100 @@ export const ReleaseDetailsLayout = ({
...
-              </Button>
-            </Flex>
+                    {formatMessage({
+                      id: 'content-releases.header.actions.release',
+                      defaultMessage: 'Release',
+                    })}
+                  </Button>
+                </CheckPermissions>
+              </Flex>
+            )
```
[the remark this answers is not in the document]
Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:296]: Let's think about it and do it in a follow up PR. I'll create a ticket to refactor.

File: packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx [hunk 4]
```diff
@@ -265,6 +313,12 @@ const ReleaseDetailsBody = () => {
   const [{ query }] = useQueryParams<GetReleaseActionsQueryParams>();
   const toggleNotification = useNotification();
   const { formatAPIError } = useAPIErrorHandler();
+  const {
+    data: releaseData,
+    isLoading: isReleaseLoading,
+    isError: isReleaseError,
+  } = useGetReleaseQuery({ id: releaseId });
```
[the remark this answers is not in the document]
Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:320]: Or maybe we should be using selectors: https://redux.js.org/tutorials/essentials/part-8-rtk-query-advanced#selecting-users-data

I'm looking into this...

[the remark this answers is not in the document]
Comment (Feranchz) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:319]: I get your point, but I don't see this as an const, instead I see this as a benefit from using RTK, in this page you sent, in this [section](https://redux.js.org/tutorials/essentials/part-8-rtk-query-advanced#cache-data-subscription-lifetimes) they talk about this, we are using the cache for this case. 

I'm not completely sure, but I think what they mention about "manually selecting user data" is to create your own logic (maybe if you don't want to use React, so you can't use the hooks), I think all these hooks we are using are doing the same behind scenes.

[the remark this answers is not in the document]
Comment (markkaylor) [reply packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx:319]: Yeah I think you are right but I really feel it's a bummer that release is possible undefined for example when we know we have the data. But we can leave it as is for now.

**Review by markkaylor — APPROVED** [review]

**Pull request merged**